zenoh/src/net/routing/pubsub.rs

```diff
@@ -1661,7 +1675,12 @@ pub fn full_reentrant_route_data(
                         } else {
                             drop(tables);
                             for (outface, key_expr, context) in route.values() {
-                                if face.id != outface.id {
+                                if face.id != outface.id
+                                    && (face.mcast_group.is_none()
```

Comment:
The rhs of `&&`:

```rust
face.mcast_group.is_none()
	|| outface.mcast_group.is_none()
	|| face.mcast_group.as_ref().unwrap() != outface.mcast_group.as_ref().unwrap()
```

can be replaced with:

```rust
match (face.mcast_group.as_ref(), outface.mcast_group.as_ref()) {
    (None,_) | (_, None)=> true,
    (Some(l), Some(r)) => l != r,
}
```
This allows to remove unnecessary `unwrap()`.

Reply:
Done.